### PR TITLE
Create Image Gallery

### LINKED ISSUE #126: Pinterest Image Gallery
The user experience for viewing images for:
* inventory homes images
* home plan images
* inspiration gallery

1. The carousel block should have the ability to use this pinterest like layout to display the images.
2. The gallery component should support auto opening if the page's url contains the #gallery hash

The inspiration gallery page should display the images on the page in this layout.

image

https://www.woodsidehomes.com/findmyhome/arizona/prescott/horizon-at-the-dells/aurora-a

### PR DESCRIPTION
**Pull request by samuelopra**
Fix #126 

Test URLs:
- Before: https://main--hubblehomes-com--aemsites.hlx.live/home-plans/plan-detail/agate
- After: https://issue-126--hubblehomes-com--aemsites.hlx.live/home-plans/plan-detail/agate

### CODE REVIEW

**Comment by bhellema** [comment]
Things are coming along nicely.  There's a few things that jumped out to me.

1. The close text should probably stay in a header so that it doesn't get lost in the images.
2. The scrolling is affecting the main page content when scrolling to the end. It even scrolls the gallery up and shows the main page content, then snaps back down.
3. The dark background feels heavy and not so fresh.  
4. I think the carousel component should be configurable to show the gallery button or not.  It seems odd that the homepage for example has a gallery button.  

Woodside has a good example to follow https://www.woodsidehomes.com/findmyhome/arizona/avondale/fire-sky-at-alamar/5024-a

**Comment by bhellema** [comment]
I think we should move the arrow buttons below the image when looking at the gallery's images one at a time.  Thoughts?

**Comment by bhellema** [comment]
Here's a screenshot of the scrolling issue on mobile.

Screenshot_20240717_203701_Chrome

**Review by bhellema — APPROVED** [review]

File: blocks/carousel/carousel.js [hunk 1]
```diff
@@ -127,10 +132,30 @@ function decorateSlideContent(col) {
   return frag;
 }
 
+function createGalleryButton() {
+  const galleryButton = document.createElement('button');
```
[the remark this answers is not in the document]
Comment (samuelopra) [reply blocks/carousel/carousel.js:139]: true, that simplifies things

**Pull request merged**